go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -184,18 +341,97 @@ func (npw *nodePortWatcher) SyncServices(services []interface{}) {
 				serviceInterface)
 			continue
 		}
-		npw.updateServiceFlowCache(service, true)
+		npw.updateServiceFlowCache(service, true, false)
 	}
 
 	npw.ofm.requestFlowSync()
-	syncSharedGatewayIptRules(services)
+	syncSharedGatewayIptRules(services, false, npw.nodeIPs)
+}
+
+// Add new Local Traffic flows if endpoints are local
+func (npw *nodePortWatcher) AddEndpoints(ep *kapi.Endpoints) {
+	name := ktypes.NamespacedName{Namespace: ep.Namespace, Name: ep.Name}
+	// Do nothing if service dosn't exist yet
```

Comment:
I was just  checking in the old code and it was [basically doing the same thing](https://github.com/openshift/ovn-kubernetes/blob/release-4.7/go-controller/pkg/ovn/endpoints.go#L46) i.e the add Endpoints function would look for a the service and if it didn't exist it would just exit. But I agree things could get messy here

Reply:
I fixed it by running a query on the API, I guess we could implement a cache and just wait until the svc existed to handle the endpoint event but I think this will work for now